src/pages/instances/forms/DuplicateInstanceForm.tsx

```diff
@@ -0,0 +1,240 @@
+    );
+  };
+
+  const formik = useFormik<DuplicateInstanceFormValues>({
+    initialValues: {
+      instanceName: instance.name + "-1",
+      project: "default",
+      allowInconsistent: false,
+      instanceOnly: false,
+      stateful: false,
+    },
+    validationSchema: Yup.object().shape({
+      instanceName: Yup.string().required(),
+    }),
+    onSubmit: (values) => {
+      duplicateInstance(values, instance)
+        .then((operation) => {
+          toastNotify.info(
+            <>
+              Creation for instance <b>{values.instanceName}</b> started.
+            </>,
+          );
+          close();
+
+          eventQueue.set(
+            operation.metadata.id,
+            async () => {
+              //Start Instance if instance created successfully.
+              const instanceToStart = fetchInstance(
+                values.instanceName,
+                values.project,
+              );
+
+              //Display starting notification
+              toastNotify.info(
+                <>
+                  Created instance <b>{values.instanceName}</b>, now starting
+                  it.
+                </>,
+              );
+
+              startInstance(await instanceToStart)
+                .then(() => {
+                  //If instance started successfully, display success message as shown in design.
+                  notifySuccess(values.instanceName);
+                })
+                .catch((e) => {
+                  toastNotify.failure(
+                    `Unable to start instance "${values.instanceName}`,
+                    e,
+                  );
+                });
+            },
+            (msg) => {
+              toastNotify.failure(
+                `Instance creation for "${values.instanceName}" failed.`,
+                new Error(msg),
+              );
+            },
+          );
+        })
+        .catch((e) => {
+          toastNotify.failure(
+            `Instance creation for "${values.instanceName}" failed.`,
+            e,
+          );
+          close();
+        });
+    },
+  });
+
+  return (
+    <Modal
+      close={close}
+      title="Duplicate Instance"
+      buttonRow={
+        <>
+          <Button
+            appearance="base"
+            className="u-no-margin--bottom"
+            type="button"
+            onClick={close}
+          >
+            Cancel
+          </Button>
+          <ActionButton
+            appearance="positive"
+            className="u-no-margin--bottom"
+            loading={formik.isSubmitting}
+            disabled={!formik.isValid}
+            onClick={() => void formik.submitForm()}
+          >
+            Duplicate & start
+          </ActionButton>
+        </>
+      }
+    >
+      <Form onSubmit={formik.handleSubmit}>
+        <Input
+          {...formik.getFieldProps("instanceName")}
+          type="text"
+          label="New instance name"
+        />
+        {isClustered ?? (
+          <Select
```

Comment:
in general, any selector that requires external data fetching should be handled with care. In the context of this form component, we should disable the selectors based on their relevant data loading states.

Reply:
Please could you clarify this and can you give an example of your suggestion?
Something I saw elsewhere in the codebase was disabling the selectors when there was only one option available (because obviously there is nothing else to choose from) is that what you are referring to?

Reply:
We should disable the fetchClusterMembers query in case clusterMembers is true.

Reply:
I would use the `isLoading` state from your various api calls